Rendera/include/cube.hpp

```diff
@@ -0,0 +1,209 @@
+#ifndef CUBE_HPP
+#define CUBE_HPP
+
+#include "vec3.hpp"
+#include "ray.hpp"
+#include "material.hpp"
+
+#define INFINITY 1E10
+
+struct HitInfo
+{
+    bool is_hitting;
+    vec3 intersection_point;
+    Material material;
+    ray Ray;
+    int recursion_depth;
+
+    HitInfo(bool is_hitting, vec3 intersection_point, Material material, ray Ray, int recursion_depth) : is_hitting{is_hitting}, intersection_point{intersection_point}, material{material}, Ray{Ray}, recursion_depth{recursion_depth} {}
```

Comment:
You don't need to define a constructor for the struct. If no constructor is defined, the compiler autogenerates one.